docs/specifications/tests/DNSSEC-TP/dnssec17.md

```diff
@@ -0,0 +1,213 @@
+
+7. Else, do (*Test Type* is "normal"):
+   1. Retrieve all name server IP addresses for the parent zone of
+      *Child Zone* using [Get-Parent-Zone] ("Parent NS IP").
+   2. For each IP address in *Parent NS IP* do:
+      1. Send the DS query over UDP to the name server IP.
+      2. If no DNS response is returned, then go to next name server
+         IP.
+      3. Else, if AA bit is not set in the DNS response, then go to
+         next name server IP.
+      3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+         go to next name server IP.
+      4. Else, if the DNS response contains at least one DS record
+         add all DS records to *DS Records*.
+
+8. If *DS Records* is empty, terminate this test case.
+
+9.  Retrieve all name server IP addresses for the *Child Zone* using
+    [Get-Del-NS-IPs] and [Get-Zone-NS-IPs] ("NS IP").
+
+10. Repeat the following steps for each name server IP address in 
+    *NS IP*:
+
+    1. Send the CDS query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server 
+          IP.
+       2. Else, if AA bit is not set in the DNS response, then go to
+          next name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+          go to next name server IP.
+       4. Else, if the DNS response contains at least one CDS record
+          in the answer section, then add the name server IP and the
+          CDS RRset to the *CDS RRsets* set. Also include any assciated
+          RRSIG records.
+    2. Send the CDNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server
+          IP.
+       2. Else, if AA bit is not set in the DNS response, then go to
+          next name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+          go to next name server IP.
+       4. Else, if the DNS response contains at least one CDNSKEY
+          record in the answer section, then add the name server IP and
+          the CDNSKEY RRset from the answer section to the 
+          *CDNSKEY RRsets* set. Also include any assciated RRSIG records.
+    3. Send the DNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server
+          IP.
+       2. Else, if AA bit is not set in the DNS response, then go to
+          next name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then
+          go to next name server IP.
+       4. Else, if the DNS response contains at least one DNSKEY
+          record in the answer section, then add the name server IP and
+          the CDNSKEY RRset from the answer section to the 
+          *DNSKEY RRsets* set.
+    4. Go to next name server IP.
+
+11. If neither of the *CDS RRsets* and *CDNSKEY RRsets* sets,
+    respectively, has any RRset then terminate this test case.
+
```

Comment:
If  all *CDS RRsets* and *CDNSKEY RRsets* are empty then terminate this test case.

Reply:
Will update wording.